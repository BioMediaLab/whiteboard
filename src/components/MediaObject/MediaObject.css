.MediaObject {
  display: flex;
}

.MediaObjectMedia {
  height: 50px;
  width: 50px;
  overflow: hidden;
}

.MediaObjectMedia img {
  max-width: 100%;
  height: auto;
}

.MediaObjectContent {
  flex: 1 1 100%;
  padding: 0 0 0 1rem;
}
