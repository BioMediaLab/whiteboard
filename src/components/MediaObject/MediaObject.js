import React from 'react'
import './MediaObject.css'

export default function MediaObject({ children, src }) {
  return (
    <div className="MediaObject">
      <div className="MediaObjectMedia">
        {src && <img src={src} alt="src" />}
      </div>
      <div className="MediaObjectContent">{children}</div>
    </div>
  )
}
