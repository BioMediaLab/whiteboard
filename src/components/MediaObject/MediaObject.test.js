import React from 'react'
import MediaObject from './'

describe('MediaObject', () => {
  test('can render', () => {
    const component = render(
      <MediaObject src="https://dummyimage.com/100x100/000/fff.gif">
        <p>Example Content</p>
      </MediaObject>
    )

    expect(component).toMatchSnapshot()
  })
})
