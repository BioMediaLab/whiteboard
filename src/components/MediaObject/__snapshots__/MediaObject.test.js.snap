// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaObject can render 1`] = `
<div>
  <div
    class="MediaObject"
  >
    <div
      class="MediaObjectMedia"
    >
      <img
        alt="src"
        src="https://dummyimage.com/100x100/000/fff.gif"
      />
    </div>
    <div
      class="MediaObjectContent"
    >
      <p>
        Example Content
      </p>
    </div>
  </div>
</div>
`;
